L1Trigger/L1THGCal/interface/be_algorithms/HGCalHardwareInterface.h

```diff
@@ -0,0 +1,71 @@
+#ifndef __L1Trigger_L1THGCal_HGCalHardwareInterface_h__
+#define __L1Trigger_L1THGCal_HGCalHardwareInterface_h__
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "DataFormats/L1THGCal/interface/HGCalCluster.h"
+#include "DataFormats/L1THGCal/interface/HGCalMulticluster.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+class HGCalHardwareInterface{
+  
+ public:
+  /* mapping is retrieved from: https://twiki.cern.ch/twiki/bin/viewauth/CMS/HGCALTriggerArchitectureDesignAndImplementation */
+  struct ClusterDataFormat{
+    enum CDF {ET = 0, CenterY = 8, CenterX = 20, Flags = 32, SizeY = 38, SizeX = 46, LocalMax = 54, NoTriggerCells = 56, LocalMax1RelX = 64,  LocalMax0ET = 72, LocalMax0RelY = 80, LocalMax0RelX = 88, LocalMax2RelY = 96,  LocalMax2RelX = 104,  LocalMax1ET = 112,  LocalMax1RelY = 120,  LocalMax3ET = 128, LocalMax3RelY = 136,  LocalMax3RelX = 144,  LocalMax2ET = 152 };
+  };
+  struct MultiClusterDataFormat{
+    enum MCDF {BHEFraction = 0, EEEFraction = 8, ET = 16, Z = 32, Phi = 42, Eta = 53, Flags = 64, QFlags = 76, NoTriggerCells = 88, TBA = 96, MaxEnergyLayer = 122};
+  };
+  
+  typedef std::vector<uint32_t> ClusterDataContainer;
+  typedef std::map<int,int>     DataFormatMap;
+
+  HGCalHardwareInterface(l1t::HGCalCluster * HGCC){
```

Comment:
I was skeptical about passing the objects, but I agree with you. I changed the code accordingly.